### ISSUE TITLE
eslint-disable-next-line [rule] doesn't work

### ISSUE THREAD

**Issue by shubich** [eslint/eslint#9980]
**About my environment**

* **ESLint Version: 4.15.0**
* **Node Version: 8.9.4**
* **npm Version: 5.6.0**

**I use babel-eslint parser**

<details>
<summary>Configuration</summary>

```js
// https://eslint.org/docs/user-guide/configuring

module.exports = {
  root: true,
  parserOptions: {
    parser: 'babel-eslint'
  },
  env: {
    browser: true,
  },
  // https://github.com/vuejs/eslint-plugin-vue#priority-a-essential-error-prevention
  // consider switching to `plugin:vue/strongly-recommended` or `plugin:vue/recommended` for stricter rules.
  extends: ['plugin:vue/essential', 'airbnb-base'],
  // required to lint *.vue files
  plugins: [
    'vue',
    'html'
  ],
  // check if imports actually resolve
  settings: {
    'import/resolver': {
      webpack: {
        config: 'build/webpack.base.conf.js'
      }
    }
  },
  // add your custom rules here
  rules: {
    // don't require .vue extension when importing
    'import/extensions': ['error', 'always', {
      js: 'never',
      vue: 'never'
    }],
    // disallow reassignment of function parameters
    // disallow parameter object manipulation except for specific exclusions
    'no-param-reassign': ['error', {
      props: true,
      ignorePropertyModificationsFor: [
        'state', // for vuex state
        'acc', // for reduce accumulators
        'e' // for e.returnvalue
      ]
    }],
    // allow optionalDependencies
    'import/no-extraneous-dependencies': ['error', {
      optionalDependencies: ['tests/unit/index.js']
    }],
    // allow debugger during development
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'import/prefer-default-export': 'off',
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
...
  AwesomeIcon.register({
    npm: {
      width: 24,
      height: 20,
      paths: [
        {
          // eslint-disable-next-line
          d: 'M0 7.334v8h6.666v1.332H12v-1.332h12v-8H0zm6.666 6.664H5.334v-4H3.999v4H1.335V8.667h5.331v5.331zm4 0v1.336H8.001V8.667h5.334v5.332h-2.669v-.001zm12.001 0h-1.33v-4h-1.336v4h-1.335v-4h-1.33v4h-2.671V8.667h8.002v5.331z',
        },
        {
          d: 'M10.665 10H12v2.667h-1.335V10z',
        },
      ],
    },
  });
...
```

```bash
npm run lint
```

**What did you expect to happen?**
no warnings
**What actually happened? Please include the actual, raw output from ESLint.**
D:\projects\~\Providers.vue
  39:3  error  Line 18 exceeds the maximum line length of 100  max-len

**Comment by shubich** [comment]
It's a *.vue file and there doesn't work any ignore rule like:

- /* eslint-disable */
- // eslint-disable-line
- // eslint-disable-next-line

also with `max-len`

**Comment by not-an-aardvark** [comment]
I can't reproduce this issue in the [demo](https://eslint.org/demo/#eyJ0ZXh0IjoiICBBd2Vzb21lSWNvbi5yZWdpc3Rlcih7XG4gICAgbnBtOiB7XG4gICAgICB3aWR0aDogMjQsXG4gICAgICBoZWlnaHQ6IDIwLFxuICAgICAgcGF0aHM6IFtcbiAgICAgICAge1xuICAgICAgICAgIC8vIGVzbGludC1kaXNhYmxlLW5leHQtbGluZVxuICAgICAgICAgIGQ6ICdNMCA3LjMzNHY4aDYuNjY2djEuMzMySDEydi0xLjMzMmgxMnYtOEgwem02LjY2NiA2LjY2NEg1LjMzNHYtNEgzLjk5OXY0SDEuMzM1VjguNjY3aDUuMzMxdjUuMzMxem00IDB2MS4zMzZIOC4wMDFWOC42NjdoNS4zMzR2NS4zMzJoLTIuNjY5di0uMDAxem0xMi4wMDEgMGgtMS4zM3YtNGgtMS4zMzZ2NGgtMS4zMzV2LTRoLTEuMzN2NGgtMi42NzFWOC42NjdoOC4wMDJ2NS4zMzF6JyxcbiAgICAgICAgfSxcbiAgICAgICAge1xuICAgICAgICAgIGQ6ICdNMTAuNjY1IDEwSDEydjIuNjY3aC0xLjMzNVYxMHonLFxuICAgICAgICB9LFxuICAgICAgXSxcbiAgICB9LFxuICB9KTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibWF4LWxlbiI6Mn0sImVudiI6e319fQ==). Is it possible you're using the `--no-inline-config` CLI flag?

**Comment by shubich** [comment]
![eslint](https://user-images.githubusercontent.com/29312810/36344139-1a00b182-1427-11e8-8e31-54efd03f0333.png)

`"lint": "eslint --ext .js,.vue src test/unit test/e2e/specs"`

**Comment by shubich** [comment]
Probably I catched the problem. Everything works after removing `'plugin:vue/essential'` from extends in **.eslintrc.js**. Full config you may see above. But,  can I keep this extension and avoid problems?

**Comment by AndrewGHC** [comment]
I've also experienced this issue, I believe it emerged in version 4.18.0. Unfortunately, I'm not sure as to the cause.

**Comment by bdwain** [comment]
I've also been unable to disable no-await-in-loop

**Comment by bdwain** [comment]
putting disable at the top of the file fixes the issue. it seems to have an issue if defined in a loop

**Comment by not-an-aardvark** [comment]
@AndrewGHC Are you also using `eslint-plugin-vue`?

**Comment by AndrewGHC** [comment]
@not-an-aardvark No - my plugins were orientated around React. So far my experience aligns with @bdwain's, my only issue was with no-await-in-loop and I was unable to disable it.

**Comment by not-an-aardvark** [comment]
@AndrewGHC I think your issue was also reported as https://github.com/eslint/eslint/issues/9992.

**Comment by shubich** [comment]
Still doesn't work

**Comment by platinumazure** [comment]
@not-an-aardvark Can this be closed?

**Comment by not-an-aardvark** [comment]
I'm not sure the original issue has been resolved, but I also couldn't reproduce it.

**Comment by not-an-aardvark** [comment]
Closing because this issue couldn't be reproduced.

**Comment by bdwain** [comment]
@not-an-aardvark sorry i didn't see this. This is an example of a check that was broken (at least when the issue was created. idk if it still is)

```js
    for (const foo of bar) {
      /* eslint-disable no-await-in-loop */
      await blah(foo);
      /* eslint-enable no-await-in-loop */
    }
```

**Comment by not-an-aardvark** [comment]
That should no longer be the case in the latest version.
